File: PMS Frontend/src/app/models/order.ts
import { Doctor } from "./doctor"
import { Medicine, MedicineForOrder } from "./medicine";

export class Order{
orderId?:number;
orderDate!:Date;
isPickedUp!:boolean;
amount!:number;
count!:number;
docterId!:number;
medicineId!:number;
// doctor!:Doctor
// medicine!:MedicineForOrder
}
export class UpdateOrder{
  orderId:number=0;
  orderDate:string='';
  isPickedUp:boolean=false;
  amount:number=0;
  count:number=0;
  docterId:number=0;
  medicineId:number=0;
  isConfirmed?:boolean=false;
  // doctor!:Doctor
  // medicine!:MedicineForOrder
  }

File: PMS Backend/PharmacyManagement_WebAPI/PharmacyManagement_WebAPI/Models/Order.cs
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using PharmacyManagement_WebAPI.Data;
using System.ComponentModel.DataAnnotations;

namespace PharmacyManagement_WebAPI.Models
{
    public class Order
    {
        public int OrderId { get; set; }
        [Required(ErrorMessage = "Date is requierd!")]
        [Display(Name = "Date")]
        public DateTime OrderDate { get; set; }
        public bool IsPickedUp { get; set; } = false;
        [Required(ErrorMessage = "Amount is requierd!")]
        [Display(Name = "Amount")]
        public int Amount { get; set; }
        public int Count { get; set; }
        public int DoctorId { get; set; }
       // public int AdminId { get; set; }
        public int MedicineId { get; set; }

        [ValidateNever]
        public DoctorRegistration Doctor { get; set; }
       // [ValidateNever]
       // public Admin Admin { get; set; }
        [ValidateNever]
        public Medicine Medicine { get; set; }
    }
}
